### PR TITLE
backend support for infinite scrolling through filtered thread lists

### PR DESCRIPTION
**Pull request by 2is10**
Also organizing socket handlers related to threads/notifications into new (inbox) and old (pre-inbox) groups.

I didn’t edit the old ones, except that I did remove the `"set_last_notify_read_time"` and `"get_last_notify_read_time"` handlers and started always sending `END_OF_TIME` as the last read time. That’s because the extension for some time has not used this value for anything visible to the user, and the server also does not use it. We will use it again in the future, but slightly differently.

### CODE REVIEW

**Comment by 2is10** [comment]
care to review @stkem?
sorry about some moved code. it was the only way for me to clearly make sense of the old vs. new notifications APIs.

**Comment by stkem** [comment]
No problem. Will look it over before monday.

File: kifi-backend/modules/eliza/app/com/keepit/eliza/MessagingController.scala [hunk 1]
```diff
@@ -735,93 +734,141 @@ class MessagingController @Inject() (
...
 
+  def getUnreadSendableNotificationsBefore(userId: Id[User], time: DateTime, howMany: Int): Seq[JsObject] = {
+    db.readOnly { implicit session =>
+      userThreadRepo.getUnreadSendableNotificationsBefore(userId, time, howMany)
+    }
+  }
+
   def getLatestMutedSendableNotifications(userId: Id[User], howMany: Int): Seq[JsObject] = {
-    db.readOnly{ implicit session =>
+    db.readOnly { implicit session =>
```
Comment (stkem) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/MessagingController.scala:822]: `db.readOnly` is a function and the stuff in `{}` is the argument. Personally, I find it confusing to have a space there (would be a syntax error in most languages).

**Comment by stkem** [comment]
lgtm

File: kifi-backend/modules/eliza/app/com/keepit/eliza/MessagingController.scala (hunk 1 above)
Comment (2is10) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/MessagingController.scala:822]: I’m not sure what languages you’re thinking of, but Scala’s support for trailing block arguments strikes me as most closely patterned after [Ruby’s block parameters](http://ruby-doc.com/docs/ProgrammingRuby/html/tut_containers.html#S2), and Ruby has the convention of using a space before the opening brace.

More importantly, perhaps, all examples I’ve seen from scala-lang.org (including the [`scala.collection` scaladoc](http://www.scala-lang.org/api/current/index.html#scala.collection.package)), [Twitter’s Effective Style](http://twitter.github.io/effectivescala/#Collections-Style) and [Martin Odersky’s Scala with Style talk](http://www.youtube.com/watch?v=kkTFx3-duc8) use the space before the opening brace of a block argument. Granted that I don’t have a great deal of exposure to Scala yet, I think you’re the only person I’ve seen regularly omit the space. I realize that Scala is such a flexible language that it’s nearly impossible for a team to have completely uniform Scala style conventions, but I think this one point is relatively easy to agree upon and standardize. I believe most of our codebase uses the space already.

**Comment by 2is10** [comment]
Thanks for reviewing!

**Pull request merged**